OpenEdXMobile/src/main/java/org/edx/mobile/base/FindCoursesBaseActivity.java

```diff
@@ -47,6 +54,12 @@
     private String lastClickEnrollCourseId;
     private boolean lastClickEnrollEmailOptIn;
 
+    @Inject
+    private CourseService courseService;
+
+    @Inject
+    private CourseAPI courseApi;
```

Comment:
`CourseAPI` only wraps those methods in which it performs some further processing, or injects default data. The `enrollInACourse()` method is not defined there. We should probably phase out the *API classes at some point, and use model classes with pre-populated default values to pass directly into the Retrofit methods.

> CourseService already has an object of CourseAPI in it.

It's the other way around: `CourseAPI` has an object of `CourseService` inside it.